.github/workflows/ci.yml

```diff
@@ -22,132 +22,138 @@ jobs:
 
-  build-lint-test-typescript:
+  build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        sdk: [go, java, python, typescript, dotnet]
+        include:
+          - sdk: go
+            install: ./scripts/install-go.sh
+            build: ./scripts/build-go-worker.sh
+            submodules: false
+          - sdk: java
+            install: ./scripts/install-java.sh && ./scripts/install-go.sh
+            build: ./scripts/build-java-worker.sh
+            submodules: false
+          - sdk: python
+            install: ./scripts/install-python.sh && ./scripts/install-go.sh
+            build: ./scripts/build-python-worker.sh
+            submodules: false
+          - sdk: typescript
+            install: ./scripts/install-node.sh && ./scripts/install-go.sh
+            build: ./scripts/build-typescript-worker.sh
+            submodules: true
+          - sdk: dotnet
+            install: ./scripts/install-dotnet.sh && ./scripts/install-go.sh
+            build: ./scripts/build-dotnet-worker.sh
+            submodules: true
     steps:
-      - name: Print build information
-        run: "echo head_ref: ${{ github.head_ref }}, ref: ${{ github.ref }}"
       - name: Checkout repo
         uses: actions/checkout@v4
         with:
-          submodules: 'true'
+          submodules: ${{ matrix.submodules }}
       - name: Install mise
         uses: jdx/mise-action@v2.4.4
         with:
           version: ${{ env.MISE_VERSION }}
       - name: Install Tools
-        run: |
-          ./scripts/install-node.sh
-          ./scripts/install-go.sh
-      - name: Lint TypeScript
-        run: |
-          ./scripts/lint-typescript-worker.sh
-      - name: Check TypeScript formatting
-        run: |
-          git diff --exit-code || (echo "TypeScript worker has uncommitted formatting changes after linting" && exit 1)
-      - name: Build TypeScript worker
-        run: |
-          ./scripts/build-typescript-worker.sh
+        run: ${{ matrix.install }}
+      - name: Build ${{ matrix.sdk }} worker
+        run: ${{ matrix.build }}
 
-  build-lint-test-dotnet:
+  kitchensink:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        sdk: [go, java, python, typescript, dotnet]
+        include:
+          - sdk: go
+            install: ./scripts/install-go.sh
+            submodules: false
+          - sdk: java
+            install: ./scripts/install-java.sh && ./scripts/install-go.sh
+            submodules: false
+          - sdk: python
+            install: ./scripts/install-python.sh && ./scripts/install-go.sh
+            submodules: false
+          - sdk: typescript
+            install: ./scripts/install-node.sh && ./scripts/install-go.sh
+            submodules: true
+          - sdk: dotnet
+            install: ./scripts/install-dotnet.sh && ./scripts/install-go.sh
+            submodules: true
```

Comment:
Separate jobs because:

1. the workers just log to stdout and it can be difficult to read the logs
2. easier to identify tests failures
3. parallelism